src/main/resources/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadYamlStep/help.groovy

```diff
@@ -0,0 +1,121 @@
+package org.jenkinsci.plugins.pipeline.utility.steps.conf.ReadYamlStep
+/*
+  ~ The MIT License (MIT)
+  ~
+  ~ Copyright (c) 2016 CloudBees Inc.
+  ~
+  ~ Permission is hereby granted, free of charge, to any person obtaining a copy
+  ~ of this software and associated documentation files (the "Software"), to deal
+  ~ in the Software without restriction, including without limitation the rights
+  ~ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  ~ copies of the Software, and to permit persons to whom the Software is
+  ~ furnished to do so, subject to the following conditions:
+  ~
+  ~ The above copyright notice and this permission notice shall be included in all
+  ~ copies or substantial portions of the Software.
+  ~
+  ~ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  ~ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  ~ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  ~ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  ~ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  ~ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  ~ SOFTWARE.
+ */
+p {
+    text("Reads a file in the current working directory or a String as a plain text ")
+    a(href: "http://yaml.org", target: "_blank") {
+        text("YAML")
+    }
+    text(" file. ")
+    text("It uses ")
+    a(href: "https://bitbucket.org/asomov/snakeyaml", target: "_blank") {
+        text("SnakeYAML")
+    }
+    text(" as YAML processor. ")
+    text("The returned objects are standard Java objects like List, Long, String, ...: ")
+    ul {
+		li("bool: [true, false, on, off]")
+        li("int: 42")
+        li("float: 3.14159")
+        li("list: ['LITE', 'RES_ACID', 'SUS_DEXT']")
+        li("map: {hp: 13, sp: 5}")
+    }
+}
+strong("Fields: ")
+ul {
+    li {
+        code("file: ")
+        text("Optional path to a file in the workspace to read the YAML datas from.")
+    }
+    li {
+        code("text: ")
+        text("An Optional String containing YAML formatted datas. ")
+        em {
+            text("These are added to the resulting object after ")
+            code("file")
+            text(" and so will overwrite any value already present if not a new YAML document.")
+        }
+    }
+    li {
+        code("maxAliasesForCollections: ")
+        text("Restrict the amount of aliases for collections (sequences and mappings) to avoid ")
+        a(href:"https://en.wikipedia.org/wiki/Billion_laughs_attack") {
+            text("Billion laughs attack. ")
+        }
+        text("The default is set by the YAML processor to 50, and you can override it with this parameter. ")
+        em {
+            int max_max_aliases_for_collections = org.jenkinsci.plugins.pipeline.utility.steps.conf.ReadYamlStep.MAX_MAX_ALIASES_FOR_COLLECTIONS
```

Comment:
This is why I had to change from html to jelly to groovy.

Reply:
Try calling `getStatic` from Jelly.

Reply:
@daniel-beck, just tried this using

```
            <j:getStatic var="maxMaxAliasesForCollections" field="MAX_MAX_ALIASES_FOR_COLLECTIONS" className="org.jenkinsci.plugins.pipeline.utility.steps.conf.ReadYamlStep"/>
            <j:getStatic var="maxMaxAliasesProperty" field="MAX_MAX_ALIASES_PROPERTY" className="org.jenkinsci.plugins.pipeline.utility.steps.conf.ReadYamlStep"/>
            <j:getStatic var="defaultMaxAliasesForCollections" field="DEFAULT_MAX_ALIASES_FOR_COLLECTIONS" className="org.jenkinsci.plugins.pipeline.utility.steps.conf.ReadYamlStep"/>
            <j:getStatic var="defaultMaxAliasesProperty" field="DEFAULT_MAX_ALIASES_PROPERTY" className="org.jenkinsci.plugins.pipeline.utility.steps.conf.ReadYamlStep"/>
            There is a maximum value you can set on this controller: ${maxMaxAliasesForCollections}.<br/>
            The administrator can change the max allowed value by setting the System property: ${maxMaxAliasesProperty}.<br/>
            The default can also be changed by setting the System property: ${defaultMaxAliasesProperty}<br/>
            so that no pipeline code needs to be changed.<br/>
            <j:if test="${defaultMaxAliasesForCollections.toInteger() > 0}">
                On this controller the default is set to ${defaultMaxAliasesForCollections}.
            </j:if>
```

but it failed with

```
Caused by: java.lang.ClassNotFoundException: org.jenkinsci.plugins.pipeline.utility.steps.conf.ReadYamlStep
	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	at org.eclipse.jetty.webapp.WebAppClassLoader.loadClass(WebAppClassLoader.java:538)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	at org.apache.commons.jelly.tags.core.GetStaticTag.doTag(GetStaticTag.java:106)
	... 140 more

The issue is explained in [JENKINS-26579](https://issues.jenkins-ci.org/browse/JENKINS-26579).

The workarounds look pretty ugly so shall we keep with the groovy view?

Reply:
Looks like the lesser of two evils 😦 Thanks for trying!